Инициализация ветки, добавление проверок стартовой страницы

lexa-jun opened the pull request.

lexa-jun committed:
Init commit

lexa-jun committed:
Создание отдельной папки под хранение путей

lexa-jun committed:
Добавление пути до кнопки в хедере
Написание проверки кнопки

lexa-jun committed:
Тест на проверку отображения банера на главном экране

lexa-jun committed:
Рефакторинг кода, изменение названий

lexa-jun committed:
Рефакторинг кода, изменение названий

lexa-jun committed:
Поверка на пустые поля

wergoapps commented on src/main/java/xpath/storage/BaseXpath.java:
```diff
@@ -0,0 +1,32 @@
+package xpath.storage;
```
Почему такой путь? Если по этому пути будут хранится только xpatch можно оставить наверное только папку xpath и внутри класс Base

wergoapps commented on src/main/java/xpath/storage/BaseXpath.java:
```diff
@@ -0,0 +1,32 @@
+public class BaseXpath {
+
+    public static final String LINK_TO_SITE =
+            "https://vatutinki.ru";  //  Стартовая страница
```
Во всех переменных ниже комментарии над переменными. А тут после. Может сделать однотипно?

wergoapps commented on src/main/java/xpath/storage/BaseXpath.java:
```diff
@@ -0,0 +1,32 @@
+
+    // Кнопка вверху сайта для перехода к предложению
+    public static final String BUTTON_OFFER_IN_SITE_HEADER =
+            "//h2[contains(text(), \"Выберите квартиру\")]/../../../../header/a";
+
+    // Текст на кнопке вверху сайта
+    public static final String BUTTON_TEXT_OFFER_SITE_HEADER =
+            "//h2[contains(text(), \"Выберите квартиру\")]/../../../../header/a/p";
```
../../../.. в xpatchs можно сократить

lexa-jun replied:
Получился данный путь 
//h2[contains(text(), "Выберите квартиру")]/../ancestor-or-self::div/header/a

wergoapps commented on src/main/java/xpath/storage/BaseXpath.java:
```diff
@@ -0,0 +1,32 @@
+    public static final String BUILDING_PAGE_DESCRIPTION =
+            "//h2[contains(text(), \"Заселение корпуса 2/2\")]";
+
+    public static final String Test =
```
Переменная Test?

lexa-jun replied:
Выпилено

wergoapps commented on src/test/java/xpath/storage/MainPageTest.java:
```diff
@@ -0,0 +1,58 @@
+        $(byXpath(BUTTON_OFFER_IN_SITE_HEADER))
+                .shouldBe(visible)
+                .click();
+        boolean buildingPageTransitionCheck = $(byXpath(BUILDING_PAGE_DESCRIPTION)).isDisplayed();
```
Переменная нигде больше как в assert не используется. Можно не создавать переменную, а сразу $(byXpath(BUILDING_PAGE_DESCRIPTION)).isDisplayed() закинуть в assert

wergoapps commented on src/test/java/xpath/storage/MainPageTest.java:
```diff
@@ -0,0 +1,58 @@
+           Указать полную проверку в том числе на ховер,
+           скрол странице, отображение объектов мешающих на странице.*/
+
+        // Проверка кнопки в хедере на переходы и отображение
```
Если это комментарий к тесту, то лучше его сделать над самим тестом

wergoapps commented on src/test/java/xpath/storage/MainPageTest.java:
```diff
@@ -0,0 +1,58 @@
+
+    @Test
+    public void showBannerTest() {
+        // Проверка баннера
```
Если это комментарий к тесту, то лучше его сделать над самим тестом

wergoapps commented on src/test/java/xpath/storage/MainPageTest.java:
```diff
@@ -0,0 +1,58 @@
+        $(byXpath(BANNER_DESCRIPTION_START_SALES_ZARECHNY))
+                .shouldBe(visible)
+                .shouldNotBe(empty);
+
```
Нет assert'a

wergoapps commented on src/test/java/xpath/storage/MainPageTest.java:
```diff
@@ -0,0 +1,58 @@
+        SelenideElement element = $(byXpath(Test));
+        actions().dragAndDropBy(element, -250, 0).perform();
+
+        sleep(4000);
```
Тут осталось что-то недоработанное

lexa-jun committed:
Изменение названия класса, пути

lexa-jun committed:
Правки:
Изменение ассерта

lexa-jun committed:
Правки:
Перенос комментариев

lexa-jun committed:
Правки:
Изменение пути до элементов

lexa-jun committed:
Правки:
Изменение пути до элементов

lexa-jun committed:
Правки:
Изменение пути до элементов
Исправление не нахождение атрибутов на сайти из-за скролла
Неактуальный xPath

wergoapps commented on src/test/java/ru/vatutinki/MainPageTest.java:
```diff
@@ -0,0 +1,155 @@
+                .shouldBe(visible)
+                .click();
+        Assertions.assertFalse($(byXpath(BUY_APARTMENT)).isDisplayed());
+        closeWebDriver();
```
Повторяется для каждого теста, необходимо вынести в метод в аннотацией @beforeTest
